src/Filesystem/DTO/FileSize.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace ILIAS\Filesystem\DTO;
+
+use ILIAS\Filesystem\Exception\IllegalArgumentException;
+
+/**
+ * Class FileSize
+ *
+ * This class provides the file size with additional information to
+ * remove the work to calculate the size to different unit like GiB, GB usw.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @since 5.3
+ * @version 1.0
+ */
+final class FileSize {
```

Comment:
Yes that is a good point, I think it could also fit well into the Data directory, but I am not entirely sure about it because the FileSize has a strong relation to the filesystem. But I fully agree with you in terms of the package dependencies.